### PR TITLE
Configuration of the kubernetes-backend should be able to toggle TLS verification

### PR DESCRIPTION
**Pull request by trumant**
Signed-off-by: Travis Truman <trumant@gmail.com>

## Hey, I just made a Pull Request!

Previously, when configuring the k8s plugin backend, the user was unable to decide whether or not
the kubernetes node client should verify the TLS cert presented by the kubernetes API server.

The default behavior was that the client **would not** verify the TLS cert which is an insecure default state.

This change defaults to a more secure state where absent explicit configuration the kubernetes client
**will verify the TLS cert** but also provides a configuration mechanism for users to choose to skip TLS verification.

#### :heavy_check_mark: Checklist

- [X] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [X] Added or updated documentation
- [X] Tests for new functionality and regression tests for bug fixes
- [X] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: plugins/kubernetes-backend/src/cluster-locator/ConfigClusterLocator.ts [hunk 1]
```diff
@@ -33,6 +33,10 @@ export class ConfigClusterLocator implements KubernetesClustersSupplier {
           name: c.getString('name'),
           url: c.getString('url'),
           serviceAccountToken: c.getOptionalString('serviceAccountToken'),
+          skipTLSVerify:
+            c.getOptionalBoolean('skipTLSVerify') === undefined
+              ? false
+              : c.getOptionalBoolean('skipTLSVerify'),
```
Comment (freben) [inline plugins/kubernetes-backend/src/cluster-locator/ConfigClusterLocator.ts:39]:
```suggestion
          skipTLSVerify: c.getOptionalBoolean('skipTLSVerify') ?? false,
```

File: docs/features/kubernetes/configuration.md [hunk 2]
```diff
@@ -78,6 +79,11 @@ cluster. Valid values are:
 | `serviceAccount` | This will use a Kubernetes [service account](https://kubernetes.io/docs/reference/access-authn-authz/service-accounts-admin/) to access the Kubernetes API. When this is used the `serviceAccountToken` field should also be set. |
 | `google`         | This will use a user's Google auth token from the [Google auth plugin](https://backstage.io/docs/auth/) to access the Kubernetes API.                                                                                             |
 
+##### `clusters.\*.skipTLSVerify`
+
+This determines whether or not the Kubernetes client verifies the TLS
+certificate presented by the API server.
```
Comment (freben) [inline docs/features/kubernetes/configuration.md:85]: mention what the default value is if it's not specified

**Comment by trumant** [comment]
@freben Thanks for the review!

**Review by freben — APPROVED** [review]
All good! 🚢

**Pull request merged**